src/extremeweatherbench/utils.py

```diff
@@ -403,3 +320,176 @@ def read_event_yaml(input_pth: str | Path) -> dict:
     with open(input_pth, "rb") as f:
         yaml_event_case = yaml.safe_load(f)
     return yaml_event_case
+
+
+def location_subset_point_obs(
+    df: pd.DataFrame,
+    min_lat: float,
+    max_lat: float,
+    min_lon: float,
+    max_lon: float,
+):
+    """Subset a dataframe based upon maximum and minimum latitudes and longitudes.
+
+    Arguments:
+        df: dataframe with point obs.
+        min_lat: minimum latitude.
+        max_lat: maximum latitude.
+        min_lon: minimum longitude.
+        max_lon: maximum longitude.
+
+    Returns a subset dataframe."""
+    location_subset_df = df[
+        (df["latitude"] >= min_lat)
+        & (df["latitude"] <= max_lat)
+        & (df["longitude"] >= min_lon)
+        & (df["longitude"] <= max_lon)
+    ]
+    return location_subset_df
```

Comment:
Equivalent, but slightly cleaner way of writing this code:

```suggestion
    mask = (
        (df["latitude"] >= min_lat) &
        (df["latitude"] <= max_lat) &
        (df["longitude"] >= min_lon) &
        (df["longitude"] <= max_lon)
    )
    return df[mask]
```

This avoids the temporary and makes it very clear what you're computing.